sdcm/sct_provision/region_definition_builder.py

```diff
@@ -86,7 +86,8 @@ def build_instance_definition(self, region: str, node_type: NodeTypeType, index:
                                   root_disk_size=self.params.get(mapper.root_disk_size),
                                   tags=tags,
                                   ssh_key=self._get_ssh_key(),
-                                  user_data=user_data
+                                  user_data=user_data,
+                                  instance_index=index
```

Comment:
Do we need this instance_index? This looks that is rather SCT related, which already pass `NodeIndex` tag. So maybe we can get it from this tag when SCT needs it?

Reply:
It's needed when provisioning nodes, and we only pass `InstanceDefinition` and `PricingModel` to the provisioner's `get_or_create_instance()` method. So we're getting it from SCT when creating the definitions and using them to pass that info to the VM creation code.

Reply:
I think we don't need it - we already have tag called `NodeIndex` - so this looks like duplication. See `sdcm/sct_provision/region_definition_builder.py` how we build InstanceDefinition. By the way, GCE also should have implementaion for `DefinitionBuilder` and it should be used in all places when we want to create `InstanceDefinition`.